src/main/java/org/zalando/nakadi/service/timeline/TimelineSync.java

```diff
@@ -0,0 +1,66 @@
+package org.zalando.nakadi.service.timeline;
+
+import java.io.Closeable;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Consumer;
+
+/**
+ * Interface for timeline locking.
+ * Nodes looks like:
+ * <pre>
+ * - timelines
+ *  + - lock                    lock for timeline versions synchronization
+ *  + - version: {version}      monotonically incremented long value (version of timelines configuration)
+ *  + - locked_et:
+ *     | - et_1                 Event types that are paused right now (et_1)
+ *     | - et_2                 The same goes here (et_2)
+ *  + - nodes                   nakadi nodes
+ *    + - {node1}: {version}    Each nakadi node exposes version being used on this node
+ * </pre>
+ */
+public interface TimelineSync {
```

Comment:
Maybe we could change names to make it clear what's local what's remotely coordinated.

`LocalLocking` -> `TimelineLocalSync`
`TimelineSync` -> `TimelineRemoteSync`

What do you think?